### PR TITLE
Moves app logging capture into resilient Go routine

### PR DESCRIPTION
**Pull request by ryanmoran**
The new implementation will continuously retry connecting to the logging endpoint to stream logs. Hopefully, with this retry logic, we will see fewer instances of failures on first connection to the logging subsystem in CF.

### CODE REVIEW

File: cutlass/cf.go [hunk 1]
```diff
@@ -493,6 +507,8 @@ func (a *App) DownloadDroplet(path string) error {
 
 func (a *App) Destroy() error {
 	if a.logCmd != nil && a.logCmd.Process != nil {
+		a.logCmd = nil
+		a.Stdout = nil
 		if err := a.logCmd.Process.Kill(); err != nil {
```
Comment (arjun024) [inline cutlass/cf.go:512]: Accessing `a.logCmd` would be a nil pointer dereference, right?

**Review by arjun024 — CHANGES_REQUESTED** [review]

File: cutlass/cf.go (hunk 1 above)
Comment (ryanmoran) [reply cutlass/cf.go:512]: The line immediately before this checks that `a.logCmd != nil`. I think we're safe to set that here.

Comment (arjun024) [reply cutlass/cf.go:512]: @ryanmoran I was referring to line `512` on which I posted the comment.
It'll end up as `nil.Process.Kill()` due to `a.logCmd` set to `nil` two lines up.

Comment (ryanmoran) [reply cutlass/cf.go:512]: oh jeez. what the hell.

Comment (ryanmoran) [reply cutlass/cf.go:512]: ok. moved to after the `Process.Kill`

**Review by arjun024 — APPROVED** [review]

**Pull request merged**